Add Redis docs (#10693)
gravitational/teleport: docs/pages/database-access/guides/gui-clients.mdx

![DataGrip connection options](../../../img/database-access/guides/sqlserver/datagrip-connection@2x.png)

Click OK to connect.

## Redis Insight

<Admonition type="note">
  Teleport's Redis Insight integration only supports Redis standalone instances.
</Admonition>

After opening Redis Insight click `ADD REDIS DATABASE`.

![Redis Insight Startup Screen](../../../img/database-access/guides/redis/redisinsight-startup.png)

> Click `Add Database Manually`. Use `127.0.0.1` as the `Host` and port printed by `tsh proxy db`.

Reviewer: I'm not sure what the second sentence means. Does `tsh proxy db` return a port to include in a Redis Insight field? If so, we should change this to:

```markdown
Run the following command to start a proxy that you can use to connect
to your Redis instance:

\`\`\`code (remove the escapes)
$ tsh proxy db
\`\`\`

Copy the port returned from this command and use it for the `Port`
field. Use `127.0.0.1` for the `Host` field.

Author: This is not that simple.  First you need to login to the database. Then call the `tsh db proxy`. This is explained at the top of the page https://goteleport.com/docs/database-access/guides/gui-clients/#get-connection-information. I don't think it's needed to repeat it here. I can add a link to the top of that page to make it more clear.

> <Admonition type="note">

Reviewer: Should this Admonition be regular text that comes before the "Click..." instruction above? Otherwise, readers will need to follow these instructions out of order.

Author: I'll move it.

>   Remember to login to your Redis instance with Redis user first by using:

Reviewer: ```suggestion
  Remember to log in to your Redis instance with a Redis user first by using:
```

  `tsh db login --db-user=alice redis-db-name`.
</Admonition>

Provide your Redis username as `Username` and password as `Password`.

![Redis Insight Configuration](../../../img/database-access/guides/redis/redisinsight-add-config.png)

> Next tick `Use TLS` and `Verify TLS Certificates` boxes and copy the CA certificate returned by `tsh proxy db`.

Reviewer: ```suggestion
Next, check the `Use TLS` and `Verify TLS Certificates` boxes and copy the CA certificate returned by `tsh proxy db`.
```

> Next, check the `Use TLS` and `Verify TLS Certificates` boxes and copy the CA certificate returned by `tsh proxy db`.

Reviewer: We should include the instructions to copy the certificate when we first tell the reader to run `tsh proxy db`. Otherwise, they may have cleared the terminal at this point. (Unless the proxy runs in the foreground? I haven't run this command myself.)

Reviewer: That's helpful, thanks. To make this clearer, I think we should replace line 210 with something like the following (removing the code fence escapes):

```markdown
Run the following command to start the Database Access proxy for your
Redis instance.

\`\`\`code
$ tsh proxy db redis 
Started DB proxy on 127.0.0.1:51823
Use following credentials to connect to the redis proxy: 
  ca_file=path/to/example.com.pem 
  cert_file=path/to/example.com/redis-x509.pem
  key_file=path/to/example.com/bob
\`\`\`

In Redis Insight, click `Add Database Manually`. Use `127.0.0.1` as the
`Host`. Copy the port printed by `tsh proxy db` and use it for the 
`Port` field.
```

Copy the private key and certificate to corresponding fields.

Click `Add Redis Database`.

![Redis Insight TLS Configuration](../../../img/database-access/guides/redis/redisinsight-tls-config.png)

> Congratulation. You have just connected to your Redis instance.

Reviewer: ```suggestion
Congratulations! You have just connected to your Redis instance.
```
